packages/types/src/api/cache.ts

```diff
@@ -0,0 +1,7 @@
+import {ApiHttpStatusCode} from './status-code'
+
+export type ApiResponseRecordWithCache<T> =
+  | [StatusCode: ApiHttpStatusCode.Ok, Record: T, ETag: string, MaxAge: number]
+  | [StatusCode: ApiHttpStatusCode.NotModified, MaxAge: number]
```

Comment:
the reason is that on mobile it will transfer "keys", like 200 "status", which is not necessary, so that's why using a positional array. so we save users' bandwidth.